code/modules/mob/mob_movement.dm

```diff
@@ -315,6 +315,10 @@
 	return 0
 
 /mob/get_spacemove_backup(movement_dir)
+	if(movement_dir == 0 && isliving(src))
+		var/mob/living/me = src
+		if(me.has_status_effect(STATUS_EFFECT_UNBALANCED))
+			return
```

Comment:
Could you put this behavior on a `/mob/living` override instead, to avoid what's essentially an `istype(src)`?